homeassistant/helpers/trigger_template_entity.py

```diff
@@ -239,7 +239,11 @@ def _process_manual_data(self, value: Any | None = None) -> None:
         Ex: self._process_manual_data(payload)
         """
 
-        self.async_write_ha_state()
+        with contextlib.suppress(ValueError):
+            # If state is not valid this silently continue
+            # Faulty states should not be handled here but later
+            # in the respective platforms state update
```

Comment:
I don't understand this comment, how can integrations deriving entity classes from `ManualTriggerEntity` handle the error later?

Also, why is the state written here, but also in most cases immediately after calling `_process_manual_data` ?

Reply:
Suppressing the error is only for not breaking the `this` variable for templating so if it can't write state it would calculate any potential templates based on the previous state.

We need to write state as it's used by the `this` variable and we don't have a coordinator or anything else that is helping with this.
The problem is now if the writing raises `ValueError` any other template for `availability` will not run as it raises before those are evaluated.

I see two possible solutions
1. As this PR suppress and use previous state
2. Suppress and create a manual state, perhaps just set it to `None`

Reply:
I did some more testing and I think the second solution is correct, e.g. use a `None` value as the current state. But there might be some other hidden issues because I repeated my tests with a template sensor and this shows different behavior:

(1) The template sensor does not evaluate the value if availability evaluates to false. No errors from the value (like division by zero in test4) appear in the logs. This appears to be the correct behavior.

(2) For the template sensor, a `None` value with availability==true results in an `unknown` value, rest just throws an exception because the `None` value is somehow converted to a string (`'None'`) and then to a float value (`could not convert string to float: 'None'`).

I used the following test setup. It shows that the template sensor ignores any value if availability==false and if availability==true, it converts `None` to `unknown`. The rest sensor only works if the value is a numeric value, in any other case it throws an error.
```yaml
rest:
  - resource: http://192.168.1.110:3001/api/status-page/heartbeat/default
    scan_interval: 30
    sensor:
      - unique_id: id_test1_rest
        name: test1_rest
        value_template: '{{ None }}'
        availability: '{{ false }}'
        unit_of_measurement: 'ms'
        state_class: measurement
      - unique_id: id_test2_rest
        name: test2_rest
        value_template: '{{ None }}'
        availability: '{{ true }}'
        unit_of_measurement: 'ms'
        state_class: measurement
      - unique_id: id_test3_rest
        name: test3_rest
        value_template: '{{ 0 }}'
        availability: '{{ false }}'
        unit_of_measurement: 'ms'
        state_class: measurement
      - unique_id: id_test4_rest
        name: test4_rest
        value_template: '{{ 0/0 }}'
        availability: '{{ false }}'
        unit_of_measurement: 'ms'
        state_class: measurement
template:
  - sensor:
      - unique_id: id_test1_template
        name: test1_template
        state: '{{ None }}'
        availability: '{{ false }}'
        unit_of_measurement: 'ms'
        state_class: measurement
      - unique_id: id_test2_template
        name: test2_template
        state: '{{ None }}'
        availability: '{{ true }}'
        unit_of_measurement: 'ms'
        state_class: measurement
      - unique_id: id_test3_template
        name: test3_template
        state: '{{ 0 }}'
        availability: '{{ false }}'
        unit_of_measurement: 'ms'
        state_class: measurement
      - unique_id: id_test4_template
        name: test4_template
        state: '{{ 0/0 }}'
        availability: '{{ false }}'
        unit_of_measurement: 'ms'
        state_class: measurement
```
With this result:
![grafik](https://github.com/home-assistant/core/assets/31872658/72882d95-1c8d-4b59-b70c-37e24272d098)

And errors:
```
sensor.test1_rest: ValueError: Sensor sensor.test1_rest has device class 'None', state class 'measurement' unit 'ms' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: 'None' (<class 'str'>)
sensor.test2_rest: ValueError: Sensor sensor.test2_rest has device class 'None', state class 'measurement' unit 'ms' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: 'None' (<class 'str'>)
sensor.test4_rest: ZeroDivisionError: division by zero
```